test/unit-test/ota_utest.c

```diff
@@ -966,10 +1042,56 @@ void test_OTA_ReceiveFileBlockComplete()
     TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
 }
 
+void test_OTA_ReceiveFileBlockCompleteHttp()
+{
+    OtaEventMsg_t otaEvent = { NULL, OtaAgentEventReceivedFileBlock };
+    uint8_t pFileBlock[ OTA_FILE_BLOCK_SIZE ] = { 0 };
+    int remainingBlocks = OTA_TEST_FILE_SIZE;
+    int idx = 0;
+
+    pOtaJobDoc = JOB_DOC_HTTP;
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForFileBlock, OTA_GetState() );
+
+    /* Set the event send interface to a mock function that allows events to be sent continuously
+     * because we're receiving multiple blocks in this test. */
+    otaInterfaces.os.event.send = mockOSEventSend;
+
+    /* Fill the file block. */
+    for( idx = 0; idx < sizeof( pFileBlock ); idx++ )
+    {
+        pFileBlock[ idx ] = idx % UINT8_MAX;
+    }
+
+    while( remainingBlocks > OTA_FILE_BLOCK_SIZE )
+    {
+        otaEvent.pEventData = &eventBuffer;
+        memcpy( otaEvent.pEventData->data, pFileBlock, OTA_FILE_BLOCK_SIZE );
+        otaEvent.pEventData->dataLength = OTA_FILE_BLOCK_SIZE;
+
+        OTA_SignalEvent( &otaEvent );
+        otaWaitForEmptyEvent();
+        TEST_ASSERT_EQUAL( OtaAgentStateWaitingForFileBlock, OTA_GetState() );
+
+        /* TODO, statistics is now broken. Need to fix it to test OTA_GetPacketsReceived
```

Comment:
Can we fix this in this PR? I don't think we should have todo statements in code that is being merged into the main branch.

Reply:
This is still broken, I just tried, all statistics are 0. Since it's in `ota.c`, I think it's better to fix in another PR.